src/interface/viewport.c

```diff
@@ -2636,8 +2636,216 @@ void store_interaction_info(paint_struct *ps)
  *
  *  rct2: 0x00679074
  */
-void sub_679074(rct_drawpixelinfo *dpi, int imageId, int x, int y)
+void sub_679074(rct_drawpixelinfo *dpi, int imageId, sint16 x, sint16 y)
 {
+	rct_g1_element image = g1Elements[imageId & 0x7FFFF];
```

Comment:
Can also be a g2 element but thats for new OpenRCT2 elements none of which currently call into this function.